src/main/java/prgms/spring_week1/domain/customer/model/embeddedType/Email.java

```diff
@@ -0,0 +1,49 @@
+package prgms.spring_week1.domain.customer.model.embeddedType;
+
+import java.util.regex.Pattern;
+
+public class Email {
+    private final int MINIMUM_EMAIL_LENGTH = 4;
+    private final int MAXIMUM_EMAIL_LENGTH = 50;
+    private final String EMAIL_FORMAT_CHECK_REGEX = "\\b[\\w\\.-]+@[\\w\\.-]+\\.\\w{2,4}\\b";
+
+    private final String address;
+
+    public Email(String address) {
+        checkAddress(address);
+        this.address = address;
+    }
+
+    private void checkAddress(String address){
+        checkAddressIsNull(address);
+        checkAddressLength(address);
+        checkAddressFormat(address);
+    }
+
+    private void checkAddressIsNull(String address) {
+        boolean addressIsNull = address == null;
+
+        if (addressIsNull) {
+            throw new IllegalArgumentException("이메일은 필수 입력값입니다.");
+        }
```

Comment:
차라리 아래가 더 직관적인 것 같아요 

```suggestion
        if (address == null) {
            throw new IllegalArgumentException("이메일은 필수 입력값입니다.");
        }
```

Reply:
해당 부분도 고민해보겠습니다 !
조건문 인자에 연산자가 들어가는  것을 지양하기 위해 따로 변수를 만들어서 대입하였습니다.